### PR TITLE
[full-ci] feat: cloud import conflict handling

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Adds conflict handling to the recently added [cloud import feature](https://github.com/owncloud/web/pull/9150).

The import process has been refactored and moved to the `HandleUpload` Uppy plugin in the course of this. This also introduces a quota check as well as the possibility to have the directory tree created. On top of that it also gathers most of the upload logic in one place instead of having it spread all over the place like before. As a result, the custom drop plugin and the `useUploadHelpers` composables were removed.

## Technical detail
Let me try to give a little technical detail to make it easier to understand the changes. How the upload worked until now: An upload automatically started as soon as files were added to the Uppy upload queue. However, we need to process and check a few things before uploading:

* convert the source files to `UppyResource`s
* is enough quota available?
* are the name conflicts? How to handle them if so?
* do we need to create directories?
* do we have failed files from a previous upload that need to be retried?

Because these things need to happen *before* an upload we needed to hook into the process before files were added to Uppy. The way we achieved this always depended on the upload source. For Drag&Drop e.g. we needed to expand the Uppy `DropPlugin`. For the newly added cloud import we needed to implement a pre processor plugin. The 2 main issues with this approach:

1. we may need to add more individual logic when adding more upload sources in the future.
2. upload-related logic is spread all over the place.

Now for this PR and what it does: adding files to the Uppy upload queue does not automatically start the upload - we need to do that manually. This gives us the possibility to hook up *after* all files have been added to Uppy, then go through the above mentioned process steps and then start the upload. That is achieved by one single plugin that always hooks up after files have been added to Uppy, no matter the upload source. That eventually solves both issues we had before.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>acceptance oC10</strong> https://drone.owncloud.com/owncloud/web/36376/23/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesActionMenu-versions_feature-L74.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>acceptance oCIS</strong> https://drone.owncloud.com/owncloud/web/36376/69/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L152.png
  

### webUIUpload-upload_feature-L22.png
  

### webUIUpload-upload_feature-L56.png
  

### webUIUpload-upload_feature-L70.png
  


</p></details>

File: packages/web-app-files/src/uppyUploadPlugin.ts [hunk 1]
```diff
@@ -118,19 +188,237 @@ export class HandleUpload extends BasePlugin {
...
+
+    // name conflict handling
+    if (this.conflictHandlingEnabled) {
+      const confictHandler = new ResourceConflict(this.store, this.language)
+      const conflicts = confictHandler.getConflicts(this.filesToUpload)
+      if (conflicts.length) {
+        const dashboard = document.getElementsByClassName('uppy-Dashboard')
+        if (dashboard.length) {
+          ;(dashboard[0] as HTMLElement).style.display = 'none'
+        }
```
Comment (dschmidt) [inline packages/web-app-files/src/uppyUploadPlugin.ts:380]: eeeks :-\

File: packages/web-app-files/src/uppyUploadPlugin.ts [hunk 2]
```diff
@@ -1,36 +1,74 @@
...
  *
- * It basically does the same as the inputFilesToUppyFiles method from the upload helpers.
- * TODO: Remove the upload helpers and use this plugin for upload handling in general.
+ * 1. convert input files to uppy resources
+ * 2. check quota if spaces are enabled
+ * 3. handle potential naming conflicts
+ * 4. create directory tree if needed
+ * 5. start upload
  */
 export class HandleUpload extends BasePlugin {
```
Comment (dschmidt) [inline packages/web-app-files/src/uppyUploadPlugin.ts:43]: I think it's a bit unfortunate that the file name does not resemble the class name at all.. can we streamline this more?

**Review by dschmidt — COMMENTED** [review]
craaaazy stuff. love it - just a few nitpicks

**Review by dschmidt — APPROVED** [review]
craaaazy stuff. love it - just a few nitpicks

File: packages/web-runtime/src/composables/upload/useUpload.ts [hunk 3]
```diff
@@ -27,6 +22,8 @@ export interface UppyResource {
   data: Blob
   meta: {
     // IMPORTANT: must only contain primitive types, complex types won't be serialized properly!
```
[the remark this answers is not in the document]
Comment (dschmidt) [reply packages/web-runtime/src/composables/upload/useUpload.ts:24]: no action time - just a question and something we could investigate at some point

**Review by kulmann — APPROVED** [review]
oh wow, christmas came early this year 🎅 😍

**Pull request merged**